TSRM/TSRM.c

```diff
@@ -170,6 +177,11 @@ TSRM_API void tsrm_shutdown(void)
 {
 	int i;
 
+	if (!in_main_thread) {
+		/* ensure singleton */
+		return;
+	}
+
 	if (tsrm_tls_table) {
```

Comment:
as we already have a in_main_thread guarder, this check should also not that necessary, right?